Source/Core/loadWithXhr.js

```diff
@@ -153,12 +153,20 @@ define([
         xhr.onload = function() {
             if (xhr.status >= 200 && xhr.status < 300) {
                 if (defined(xhr.response)) {
-                    deferred.resolve(xhr.response);
+                    if (responseType === 'json' && typeof xhr.response === 'string') { //IE doesn't support responseType = 'json'
```

Comment:
I spent a bunch of time verifying browser behaviors here and ended up refacotring `xhr.onload` to incorporate your changes but also better separate "old browser" behavior from modern browsers.  Can you replace xhr.onload with the below?

```javascript
        xhr.onload = function() {
            if (xhr.status < 200 || xhr.status >= 300) {
                deferred.reject(new RequestErrorEvent(xhr.status, xhr.response, xhr.getAllResponseHeaders()));
                return;
            }

            var response = xhr.response;
            var browserResponseType = xhr.responseType;

            //All modern browsers will go into either the first if block or last else block.
            //Other code paths support older browsers that either do not support the supplied responseType
            //or do not support the xhr.response property.
            if (defined(response) && (!defined(responseType) || (browserResponseType === responseType))) {
                deferred.resolve(response);
            } else if ((responseType === 'json') && typeof response === 'string') {
                try {
                    deferred.resolve(JSON.parse(response));
                } catch (e) {
                    deferred.reject(e);
                }
            } else if ((browserResponseType === '' || browserResponseType === 'document') && defined(xhr.responseXML) && xhr.responseXML.hasChildNodes()) {
                deferred.resolve(xhr.responseXML);
            } else if ((browserResponseType === '' || browserResponseType === 'text') && defined(xhr.responseText)) {
                deferred.resolve(xhr.responseText);
            } else {
                deferred.reject(new RuntimeError('Invalid XMLHttpRequest response type.'));
            }
        };
```

This will also break a couple of your tests, but that's because your tests are depending on non-standard XHR behavior that is a result of your mocking the XHR object.  I spoke with @shunter offline and he disagreed with me and felt that mocking XHR was okay, but we need to make sure the behavior matches what modern browsers do (and also matches old browser behavior in tests that exercise it.).

So update your tests to work with the above code and then I'll take another look.